### DISCUSSION TITLE
Volume returned by ib.reqHistoricalData in lots or in shares?

### DISCUSSION THREAD

**Discussion by noob974** [ib-api-reloaded/ib_async#86] [Q&A] [answered]
Please forgive my poor English. And please forgive me if I'm terribly outdated.

I have been using ib_insync 0.9.7 version before. I have some code like this:
```
from ib_insync import *  # now changed to ib_async
symbol = "AAPL"
contract_list = ib.reqContractDetails(Stock(symbol, exchange="SMART", currency="USD"))
contract = contract_list[0].contract
data = ib.reqHistoricalData(
        contract=contract,
        endDateTime="",
        durationStr='7 D',
        barSizeSetting='1 min',
        whatToShow='TRADES',
        useRTH=False,
        formatDate=2,
        keepUpToDate=False,
        timeout=120
    )
```
In the returned data, the volume is always in lots. And these codes have been working fine. 

But these days, I replaced ib_insync 0.9.7 with ib_async 1.0.3. All the codes are still running fine except the code mentioned above. The volume in the returned data is changed to units of shares. In other words, all volumes are now 100 times their original size.

With my limited ability, I could not find the corresponding explanation in the relevant documents. Did something changed in this fuction? Or has the data structure of IB changed?

Thanks for your help.

**Comment by mattsta** [comment, marked as answer]
<img width="501" alt="Screenshot 2024-10-30 at 03 55 47" src="https://github.com/user-attachments/assets/fb37d898-0322-4ee1-a3ad-4de54b8ba90b">

https://interactivebrokers.github.io/tws-api/market_data.html

> Previously, US stock size quotes were displayed in round lots (of 100 shares).
> Effective with TWS release 985 and above, the bid, ask, and last size quotes are displayed in shares instead of lots.

**Reply by noob974** [reply]
Thank you very much for your reply!
Does this mean that, in the current and future versions of ib_async, as long as I keep the "Send market data in lots for US stocks for dual-mode API clients" option unchecked, the data received by `ib.reqHistoricalData` will always be in shares?
